From b44b364b2762579776addb7371f71b77fce62eb8 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Mon, 8 Oct 2018 12:20:36 +0100
Subject: [PATCH] BCM2708_DT: Use upstreamed GPIO expander driver

The upstreamed driver for the GPIO expander has a different compatible
string. Change the relevant Device Tree files to match.

See: https://github.com/raspberrypi/linux/issues/2704

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts | 2 +-
 arch/arm/boot/dts/bcm2710-rpi-3-b.dts      | 2 +-
 arch/arm/boot/dts/bcm2710-rpi-cm3.dts      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

--- a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
@@ -84,7 +84,7 @@
 
 &soc {
 	expgpio: expgpio {
-		compatible = "brcm,bcm2835-expgpio";
+		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
 		firmware = <&firmware>;
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -92,7 +92,7 @@
 	};
 
 	expgpio: expgpio {
-		compatible = "brcm,bcm2835-expgpio";
+		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
 		firmware = <&firmware>;
--- a/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-cm3.dts
@@ -52,7 +52,7 @@
 	};
 
 	expgpio: expgpio {
-		compatible = "brcm,bcm2835-expgpio";
+		compatible = "raspberrypi,firmware-gpio";
 		gpio-controller;
 		#gpio-cells = <2>;
 		firmware = <&firmware>;
